collector/agent_poller.go

```diff
@@ -383,12 +415,12 @@ func getAgentState(agentIp string, stats chan<- StatsEvent) (*AgentState, error)
 		executorAppNames: executorAppNames}, nil
 }
 
-func getJsonFromAgent(agentIp string, urlPath string, testFileFlag *string, stats chan<- StatsEvent) (*jason.Value, error) {
+func (a *Agent) getJsonFromAgent(urlPath string, testFileFlag *string, stats chan<- StatsEvent) (*jason.Value, error) {
 	stats <- MakeEvent(AgentQuery)
 	var rawJson []byte = nil
 	var err error = nil
 	if len(*testFileFlag) == 0 {
-		endpoint := fmt.Sprintf("http://%s:%d%s", agentIp, *agentPortFlag, urlPath)
+		endpoint := fmt.Sprintf("http://%s:%d%s", a.AgentIP, a.Port, urlPath)
```

Comment:
this is also not the right way of doing it, can we encode a URL with https://golang.org/pkg/net/url/#URL
there is an example